### PR TITLE
Randomization of tests with `DETACH`/`ATTACH` table

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Randomly execute `DETACH`/`ATTACH` queries for all tables that are used in query from test before its execution.

### CODE REVIEW

File: tests/queries/0_stateless/00626_replace_partition_from_table.sql [hunk 1]
```diff
@@ -1,4 +1,5 @@
--- Tags: no-parallel
+-- Tags: no-parallel, no-random-detach
+-- Tag no-random-detach: it's a bug, will be fixed in https://github.com/ClickHouse/ClickHouse/pull/41145
```
Comment (CurtizJ) [inline tests/queries/0_stateless/00626_replace_partition_from_table.sql:2]: The bug is that after `TRUNCATE` or `DROP PARTITION` a new part with the same name (and path) may appear, but marks/uncompressed cache is not dropped, which leads to reading of wrong data. Most likely will be fixed after #41145.

File: tests/queries/0_stateless/01459_manual_write_to_replicas_quorum.sh [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
-# Tags: replica, no-replicated-database, no-parallel
+# Tags: replica, no-replicated-database, no-randon-detach
 # Tag no-replicated-database: Fails due to additional replicas or shards
+# Tag no-random-detach: it's a bug, will be fixed in https://github.com/ClickHouse/ClickHouse/pull/42509
```
Comment (CurtizJ) [inline tests/queries/0_stateless/01459_manual_write_to_replicas_quorum.sh:4]: #42509

File: tests/queries/0_stateless/01149_zookeeper_mutation_stuck_after_replace_partition.sql [hunk 3]
```diff
@@ -1,4 +1,7 @@
--- Tags: zookeeper
+-- Tags: zookeeper, no-random-detach
+-- Tag no-random-detach: "The local set of parts of table
+-- doesn't look like the set of parts in ZooKeeper"
+-- after DROP PARTITION and restart https://github.com/ClickHouse/ClickHouse/issues/37664
```
Comment (CurtizJ) [inline tests/queries/0_stateless/01149_zookeeper_mutation_stuck_after_replace_partition.sql:4]: #37664

File: tests/queries/0_stateless/00626_replace_partition_from_table.sql (hunk 1 above)
Comment (qoega) [reply tests/queries/0_stateless/00626_replace_partition_from_table.sql:2]: Is it?

File: tests/queries/0_stateless/01459_manual_write_to_replicas_quorum.sh (hunk 2 above)
Comment (qoega) [reply tests/queries/0_stateless/01459_manual_write_to_replicas_quorum.sh:4]: This one should be fixed in #44628

**Comment by alexey-milovidov** [comment]
What a nice feature!

**Comment by alexey-milovidov** [comment]
We need this!

**Pull request closed without merging**